### ISSUE TITLE
Using timm scheduler, metric in `lr_scheduler_step` will always be `None`

### ISSUE THREAD

**Issue by timothylimyl** [Lightning-AI/lightning#13260]
Hi, was wondering whether does anyone else has this issue?

When I change my scheduler to a pytorch scheduler, it suddenly works and I can see that the `metric` has value  in `lr_scheduler_step()`. Snippet of code in `LightningModule`:

```
    def configure_optimizers(self):

        return [self.opt], [
            {
                "scheduler": self.sch,
                "interval": "epoch",
                "monitor": "val_loss",
                "frequency": 1,
            }
        ]

    def lr_scheduler_step(self, scheduler, optimizer_idx, metric):

        # Some PL bug for Metric, metric is none and cannot be passed even after it is logged. But it works for torch reducelr.
        # As of now, cannot use LR Scheduler that is based off metric in timm schedulers.

        scheduler.step(
            epoch=self.current_epoch, metric=metric
        )  # note to change argument from metric to metrics when using pytorch official schedulers.

```

I cannot figure out why metric is `None` when using a timm scheduler. All logging code remains the same. 

Snippet in validation (val_loss is being logged):
```
    def eval_step(self, batch, batch_idx, prefix: str):
        images, target = batch
        output = self.model(images)
        loss_val = F.cross_entropy(output, target)

        # update metrics
        self.eval_acc1(output, target)
        self.eval_acc5(output, target)

        self.log(f"{prefix}_loss", loss_val)
        self.log(
            f"{prefix}_acc1",
            self.eval_acc1,
        )
        self.log(
            f"{prefix}_acc5",
            self.eval_acc5,
        )

    def validation_step(self, batch, batch_idx):
        return self.eval_step(batch, batch_idx, "val")
```
Reading around, technically timm scheduler should work without any problems, all the timm scheduler without the need of metric works for me but scheduler like plateau will require metric.

cc @carmocca @awaelchli @borda @ninginthecloud @rohitgr7

**Comment by timothylimyl** [comment]
Hi @carmocca , happy to work on a PR but I do need some pointers from more experienced developers on where to make the changes.

**Comment by carmocca** [comment]
I believe this was fixed with https://github.com/Lightning-AI/lightning/pull/13838. You will need to add the key `reduce_lr_on_plateau` in `configure_optimizers` when the timm scheduler takes a monitor. For example:
```python
lr_scheduler = {"reduce_on_plateau": True, "scheduler": CustomReduceLROnPlateau(), "monitor": "my_loss"}
```
Note the required `monitor` key too
